dart/dynamics/Entity.h

```diff
@@ -63,11 +63,23 @@ class Shape;
 /// may have different policies about how/if their reference frame or name
 /// can be changed. Use the Detachable class to create an Entity whose reference
 /// Frame can be changed arbitrarily.
-class Entity
+class Entity : public common::Subscription
 {
 public:
   friend class Frame;
 
+  enum Notification {
```

Comment:
Or we can change the parameter type to the enumeration like:
```cpp
void Subscription::sendNotification(Notification _notice) {}
void Subscriber::receiveNotification(Subscription* _subscription, Notification _notice) {}
```
But this is not extensible in terms of Notification type.

If we need to choose one of the advantages of strongly typed enumerations and avoiding grossness of casting, we should go for the avoiding grossness of casting.

By the way, what do you think about class `Event` I mentioned in other reply. It is composition design while Subscription is inheritance. Generally [composition is more flexible than inheritance], and `Event` can prevent unnecessary notifications of a notice for the subscriber that is not interested in the notice.